src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1187 @@
+    return this._getApproveTokenRequest(
+      params,
+      this.teleporterAddresses.l1Teleporter
+    )
+  }
+
+  /**
+   * Approve tokens for teleportation. The tokens will be approved to be spent by the `L1Teleporter`
+   */
+  public async approveToken(
+    params: TokenApproveParams,
+    l1Signer: Signer
+  ): Promise<ethers.ContractTransaction> {
+    await this._checkL1Network(l1Signer)
+
+    const approveRequest = await this.getApproveTokenRequest(params)
+
+    return l1Signer.sendTransaction(approveRequest)
+  }
+
+  /**
+   * Get a tx request to deposit tokens to L3 through the L1Teleporter contract.
+   */
+  public async getDepositRequest(
+    params: Erc20DepositRequestParams,
+    l1Signer: Signer,
+    l2Provider: Provider,
+    l3Provider: Provider
+  ): Promise<Required<Pick<TransactionRequest, 'to' | 'data' | 'value'>>> {
+    await this._checkL1Network(l1Signer)
+    await this._checkL2Network(l2Provider)
+    await this._checkL3Network(l3Provider)
+
+    const gasParams = await this._populateGasParams(
+      params,
+      l1Signer.provider!,
+      l2Provider,
+      l3Provider
+    )
+
+    const teleporter = L1Teleporter__factory.connect(
+      this.teleporterAddresses.l1Teleporter,
+      l1Signer
+    )
+
+    const calldata = teleporter.interface.encodeFunctionData('teleport', [
+      {
+        l1Token: params.erc20L1Address,
+        l1l2Router: this.l2Network.tokenBridge.l1GatewayRouter,
+        l2l3Router: this.l3Network.tokenBridge.l1GatewayRouter,
+        to: params.to || (await l1Signer.getAddress()),
+        amount: params.amount,
+        gasParams,
+      },
+    ])
+
+    const adjustedL1GasPrice = this._percentIncrease(
+      params.overrides?.l1GasPrice?.base ||
+        (await l1Signer.provider!.getGasPrice()),
+      params.overrides?.l1GasPrice?.percentIncrease ||
+        this.defaultGasPricePercentIncrease
+    )
+
+    const calculatedGasCosts = await teleporter.calculateRetryableGasCosts(
+      this.l2Network.ethBridge.inbox,
+      adjustedL1GasPrice,
+      gasParams
+    )
+
+    return {
+      to: this.teleporterAddresses.l1Teleporter,
+      data: calldata,
+      value: calculatedGasCosts.total,
+    }
+  }
+
+  /**
+   * Deposit tokens to L3 through the L1Teleporter contract.
+   */
+  public async deposit(
+    params: Erc20DepositRequestParams,
+    l1Signer: Signer,
+    l2Provider: Provider,
+    l3Provider: Provider
+  ): Promise<L1ContractCallTransaction> {
+    const txRequest = await this.getDepositRequest(
+      params,
+      l1Signer,
+      l2Provider,
+      l3Provider
+    )
+
+    return L1TransactionReceipt.monkeyPatchContractCallWait(
+      await l1Signer.sendTransaction(txRequest)
+    )
+  }
+
+  /**
+   * Get the status of a deposit given an L1 tx receipt. See `Erc20DepositStatus` interface for more info on the return type.
+   *
+   * Note: This function does not verify that the tx is actually a deposit tx.
+   */
+  public async getDepositStatus(
```

Comment:
Generally we've put the status methods on the messages themselves. In this case we dont have an l1toL3 message, I'm not sure if it makes sense to have one or not, what do you think?

If we do keep this functions here I feel like they could be simplified by using the existing status functions on the underlying l1tol2Messages

Reply:
i'm not sure. i put it here initially because it was easier.

one argument though for keeping it here is that the statuses will be different for each type of L1-L3 transfer:
standard erc20 bridge: 2x L2, 1x L3 initiated by the second L2 message
relayed erc20: 1x L2, 1x L3 initiated by a relayer
ETH: 1x L2, 1x L3 initiated by L2 message

whereas with the existing L1L2 message status functions it's just a single message.

Reply:
yeh, i think it's ok to keep it - unless we make a full l1tol2message class.

However internally i think we should use the status() functions to calculate the return of getDepositStatus